### ISSUE TITLE
Decidim Awesome : V0.8.x vs V0.9x issue with the Dockerfile and publish process/components

### ISSUE THREAD

**Issue by sylvie-mcn** [decidim-ice/decidim-module-decidim_awesome#219]
Hello !
We work with Decidim V27.2 and we try to install Awesome.

We were able to adapt the Dockerfile, so that it can receive the precompile assets command, except that we have a problem when adding the bin/rails assets:precompile command in the install-2.sh file.

Even adding the precompilation command in install-2.sh or Dockerfile.
build stops, and returns error messages ERROR in ..... Can't resolve '@rails/activestorage' ..... plus assets (js, css, etc) it can't find not.
![image](https://user-images.githubusercontent.com/96497265/232864850-70855a66-84fe-4f29-9f86-32046c527b20.png)

Also, we cannot release processes or features without having an error.

Will you publish a V0.9 release for decidim V26 and V27 soon?

Thank you !

**To Reproduce**
Decidim V0.27.2
Install Awesom V0.9 for Decidim V0.27
1.create un process
2.create a component
Publish all

I have this error for the process and this is the same for the component
![image](https://user-images.githubusercontent.com/96497265/232862131-2debb813-e39f-4711-9f04-e4ed7d74e289.png)

**Desktop (please complete the following information):**
 - OS:  windows 
 - Browser : chrome
 - Version [e.g. 27.2]

**Comment by microstudi** [comment]
I don't think you can do a GET request agains that route (publish a component). Please share your Dockerfile and your application, I'm pretty sure this has nothing to do with this plugin.

**Comment by esdrasitq** [comment]
Hi @microstudi,

@sylvie-mcn and I are working together in the same team. Just to be clear. The build was succesfully done, the Dockerfile is OK. 
So when I pass the command `bin/rails assets:precompile RAILS_ENV=Development` in the rails console, this is the error I have.

![image](https://user-images.githubusercontent.com/91143396/232873917-da6e2be0-6f64-41d8-a923-7f61959503e6.png)

And then, if @sylvie-mcn  try to publish a process on the website, we have the routing error displayed above. Also, in the log of the developper tools we can see the the **rails active storage is missing**.

![image](https://user-images.githubusercontent.com/91143396/232875733-4ec4a986-f014-4591-9e61-e622c057e38b.png)

Even if I try to run this command in development mode `RAILS_ENV=production bin/rails decidim_awesome:active_storage_migrations:migrate_from_carrierwave`, we still have the same problem.

I hope, this will help you understand better our issues. Any help will be appreciate. Thanks in advance.

**Comment by microstudi** [comment]
It seems that node packages aren't properly installed. Try to remove the node_modules folder and run `yarn install` or `npm ci`.
If you show your repositori I'd might be able to know more.

**Comment by esdrasitq** [comment]
@microstudi 
Thank you so much. 
After some analysis, I have identified the reason. 
In the Awesome [gemfile.lock](https://github.com/decidim-ice/decidim-module-decidim_awesome/blob/main/Gemfile.lock), activestorage version is (= 6.1.7.2). However in decidim-core module v27.2 in the [package.json file](https://github.com/decidim/decidim/blob/v0.27.2/packages/core/package.json), we can see that "@rails/activestorage": "^6.0.4". I just replace this latter by ^6.1.7 also, replace it in other package.json im my decidim folder.

Now everything works fine. :):)
![image](https://user-images.githubusercontent.com/91143396/232904794-a56b7e7b-0bb4-4a67-8bc3-5392429abc05.png)

**Comment by microstudi** [comment]
good to know, maybe we will have to change this. thanks for the explanation

**Comment by sylvie-mcn** [comment]
@microstudi
Thanks for your help !

**Comment by lawrencemcn** [comment]
Hi @microstudi 
actually just changing the active version to ^6.1.7 is not fixing the compiling issue of decidim_awesome in and out the Docker file.

Here is the trace I have for the execution of the  bundle exec rails decidim_decidim_awesome:webpacker:install in the Dockerfile:

![Warning_awesome_install](https://user-images.githubusercontent.com/105821969/235244750-66c0e975-88e5-4b31-896c-b331c1d973d7.PNG)

Is it possible that all the errors are due to the npm and node versions not being fully compatible ?

After the container is up, here is the trace for the execution of bin/rails asset:precompile --trace in the container:
There are some warnings/errors but the assets are compiled 
 
![Warning_asset_precompile_container](https://user-images.githubusercontent.com/105821969/235244964-2cdb83a4-75e0-4f64-8284-88609e257281.PNG)

Is there an example of a Dockerfile that runs Decidim with awesome after executing the commands bundle exec rails decidim_decidim_awesome:webpacker:install and bin/rails asset:precompile ?

Thank you.

**Comment by microstudi** [comment]
I think you need first make sure you are using the proper versions for npm and node (I think are 8 and 16)

**Comment by joserwan** [comment]
Hi, it looks like `Decidim.webpacker.compile` refers to something like `Webpacker::Compiler.new(Webpacker.instance)`

This seems to work:

```diff
--- a/decidim/lib/tasks/decidim_tasks.rake
+++ b/decidim/lib/tasks/decidim_tasks.rake
@@ -21,7 +21,7 @@ namespace :decidim do
     task compile: [:npm_install, :environment] do
       Webpacker.with_node_env("production") do
         ensure_log_goes_to_stdout do
-          if Decidim.webpacker.commands.compile
+          if Webpacker::Compiler.new(Webpacker.instance).compile
             # Successful compilation!
             puts "SUCCESS"
             Dir.chdir(File.join(__dir__, "../..")) do
```

Can you confirm that this is equivalent ? If so, where is defined `Decidim.webpacker` ?

**Comment by joserwan** [comment]
> I think you need first make sure you are using the proper versions for npm and node (I think are 8 and 16)

Same issue with the expected Node / NPM versions:

```bash
root@7bbadbb443af:/home/decidim/decidim-app# node -v
v16.18.1
root@7bbadbb443af:/home/decidim/decidim-app# npm -v
8.19.2
```

**Comment by microstudi** [comment]
Closing this. Reopen if still applies
